CMakeLists.txt

```diff
@@ -34,9 +34,9 @@ set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 if(NOT WIN32)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
+    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O0 -g -DDEBUG")
```

Comment:
We already have `appveyor` up and running to test compilation on Windows at each commit.  
There is no need to move the lines above within the `if` and exclude them when running on Windows.  
What's exactly the problem you are trying to solve?  
This change just breaks the build system, it doesn't look a good one honestly.